runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4863,8 +4864,23 @@ class INTERPRETER_CLASS
 //			Trc_JCL_attach_loadAgentLibrary(env, agentLibraryUTF, agentOptions, decorate);
 			const char *agentOptionsUTF = env->GetStringUTFChars(agentOptions, NULL);
 			if (NULL != agentOptionsUTF) {
+				if (J9_ARE_ANY_BITS_SET(_vm->runtimeFlags, J9_RUNTIME_ALLOW_DYNAMIC_AGENT)) {
+#if JAVA_SPEC_VERSION >= 21
+					J9JavaVM *vm = _vm;
+					/* No warning if -XX:+EnableDynamicAgentLoading is specified. */
+					if (0 > FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, VMOPT_XXENABLEDYNAMICAGENTLOADING, NULL)) {
```

Comment:
This needs to check if VMOPT_XXENABLEDYNAMICAGENTLOADING is the right most option, it could be followed by VMOPT_XXNOENABLEDYNAMICAGENTLOADING.

Reply:
This is under `J9_ARE_ANY_BITS_SET(vm->runtimeFlags, J9_RUNTIME_ALLOW_DYNAMIC_AGENT)` which ensures that there is no `VMOPT_XXNOENABLEDYNAMICAGENTLOADING` at the rightmost position.